.github/workflows/create-production-pr.yml

```diff
@@ -0,0 +1,81 @@
+name: 'Stencil Release PR Creation'
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        required: true
+        type: choice
+        description: Which version should be published?
+        options:
+          - prerelease
+          - prepatch
+          - preminor
+          - premajor
+          - patch
+          - minor
+          - major
+
+jobs:
+  create-stencil-release-pull-request:
+    name: Generate Stencil Release PR
+    runs-on: ubuntu-latest
+    steps:
+      # Log the input from GitHub Actions for easy traceability
+      - name: Log GitHub Input
+        run: |
+          echo "Version: ${{ inputs.version }}"
+        shell: bash
+
+      - name: Checkout Code
+        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+        with:
+          # A depth of 0 gets the entire git history.
+          # We need git history to generate the changelog; however, we don't know how deep to go.
+          # Since publishing is a one-off activity, just get everything.
+          fetch-depth: 0
+
+      - name: Get Core Dependencies
+        uses: ./.github/workflows/actions/get-core-dependencies
+
+      - name: Run Publish Preparation Script
+        run: npm run release.ci.prepare -- --version ${{ inputs.version }} --any-branch
+        shell: bash
+
+      - name: Log Generated Changes
+        run: git --no-pager diff
+        shell: bash
+
+      - name: Generate Version String and Branch Name
+        id: name_gen
+        run: |
+          VERSION_STR=$(jq '.version' package.json | sed s/\"//g)
+          echo "VERSION_STR=$VERSION_STR" >> "$GITHUB_OUTPUT"
+          echo "BRANCH_NAME=release/$VERSION_STR-run-${{ github.run_number }}-${{ github.run_attempt }}" >> "$GITHUB_OUTPUT"
+        shell: bash
+
+      - name: Print Version String and Branch Name
+        run: |
+          echo Version: ${{ steps.name_gen.outputs.VERSION_STR }}
+          echo Branch Name: ${{ steps.name_gen.outputs.BRANCH_NAME }}
+        shell: bash
+
+      - name: Create Git Branch
+        # Note: We'll let GitHub clean up the branch for us (this is configured in the repo settings on GitHub)
+        run: git checkout -b ${{ steps.name_gen.outputs.BRANCH_NAME }}
```

Comment:
do we need to do any error handling to ensure the branch doesn't already exist?

Reply:
I think if `$BRANCH` already exists then `git checkout -b $BRANCH` will have a non-zero exit code (looks like 128) so the workflow will fail at that point

Reply:
Yeah - it should fail. `git checkout -b $BRANCH` _will_ return 0 though when a branch already exists if there aren't any commits on the base branch, but since we're grabbing `main` (and other established branches) that shouldn't be an issue

Reply:
ok cool that makes sense to me, if we see this fail we'll need to probably do some manual intervention because something got messed up anyway (i.e. we tried to create a release and closed the PR for some reason)